apple/internal/apple_framework_import.bzl

```diff
@@ -606,3 +752,151 @@ objc_library(
 ```
 """,
 )
+
+_xcframework_import_common_attrs = {
+    "library_identifiers": attr.string_dict(
+        doc = """
+A optional key-value map of platforms to the corresponding platform IDs
+(containing all supported architectures), relative to the xcframework. The
+identifier keys should be case-insensitive variants of the values in
+`apple_common.platform`](https://docs.bazel.build/versions/5.0.0/skylark/lib/apple_common.html#platform);
+for example, `ios_device` or `ios_simulator`. The identifier values should be
+case-sensitive variants of values that might be found in the
+`LibraryIdentifier` of an `Info.plist` file in xcframework's root; for example,
+`ios-arm64_i386_x86_64-simulator` or `ios-arm64_armv7`.
+
+Passing this attribute should not be neccessary if the xcframework follows the
+standard naming convention (that is, it was created by Xcode or Bazel).
+""",
+    ),
+    "xcframework_imports": attr.label_list(
+        allow_empty = False,
+        allow_files = True,
+        mandatory = True,
+        doc = """
+The list of files under a .xcframework directory which are provided to Apple
+based targets that depend on this target.
+""",
+    ),
+    "deps": attr.label_list(
+        aspects = [swift_clang_module_aspect],
+        doc = """
+A list of targets that are dependencies of the target being built, which will
+provide headers (if the importing xcframework is a dynamic framework) and be
+linked into that target.
+""",
+        providers = [
+            [apple_common.Objc, CcInfo],
+            [apple_common.Objc, CcInfo, AppleFrameworkImportInfo],
+        ],
+    ),
+}
+
+apple_dynamic_xcframework_import = rule(
+    implementation = _apple_dynamic_xcframework_import_impl,
+    fragments = ["apple"],
+    attrs = dicts.add(_xcframework_import_common_attrs, {
+        "bundle_only": attr.bool(
+            default = False,
+            doc = """
+Avoid linking the dynamic framework, but still include it in the app. This is
+useful when you want to manually dlopen the framework at runtime.
+""",
+        ),
+    }),
+    doc = """
+This rule encapsulates an already-built dynamic xcframework. It is defined by a
+list of files in exactly one `.xcframework` directory.
+`apple_dynamic_xcframework_import` targets need to be added to library targets
+through the `deps` attribute.
+
+### Examples
+
+```starlark
+apple_dynamic_xcframework_import(
+    name = "my_dynamic_xcframework",
+    xcframework_imports = glob(["my_dynamic_framework.xcframework/**"]),
+)
+
+objc_library(
+    name = "foo_lib",
+    ...,
+    deps = [
+        ":my_dynamic_xcframework",
+    ],
+)
+```
+""",
+)
+
+# TODO: Support for adding `includes` for xcframeworks with static archives
```

Comment:
What would this need?

Reply:
Probably a Make-style variable and an additional attribute, for example:

```
includes = ["ThirdPartySDK.xcframework/$(XCFRAMEWORK_LIBRARY_ID)/Headers"],
```

In the meantime, users can use `select`.